pkg/ebpf/bytecode/runtime/log_debug_test.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package runtime
+
+import (
+	"bufio"
+	"math"
+	"os"
+	"testing"
+
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/ebpf/ebpftest"
+	manager "github.com/DataDog/ebpf-manager"
+	"github.com/DataDog/ebpf-manager/tracefs"
+	"github.com/cilium/ebpf"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/unix"
+)
+
+//go:generate $GOPATH/bin/include_headers pkg/ebpf/testdata/c/runtime/logdebug-test.c pkg/ebpf/bytecode/build/runtime/logdebug-test.c pkg/ebpf/c pkg/network/ebpf/c/runtime pkg/network/ebpf/c
+//go:generate $GOPATH/bin/integrity pkg/ebpf/bytecode/build/runtime/logdebug-test.c pkg/ebpf/bytecode/runtime/logdebug-test.go runtime
+
+func TestPatchPrintkNewline(t *testing.T) {
+	ebpftest.TestBuildMode(t, ebpftest.RuntimeCompiled, "", func(t *testing.T) {
+		cfg := ddebpf.NewConfig()
+		require.NotNil(t, cfg)
+
+		buf, err := LogdebugTest.Compile(cfg, []string{"-g", "-DDEBUG=1"}, nil)
+		require.NoError(t, err)
+		defer buf.Close()
+
+		idPair := manager.ProbeIdentificationPair{
+			EBPFFuncName: "logdebugtest",
+			UID:          "logdebugtest",
+		}
+		mgr := &manager.Manager{
+			Probes: []*manager.Probe{
+				{
+					ProbeIdentificationPair: idPair,
+					XDPAttachMode:           manager.XdpAttachModeSkb,
+					IfIndex:                 1,
+				},
+			},
+		}
+
+		opts := manager.Options{
+			RLimit: &unix.Rlimit{
+				Cur: math.MaxUint64,
+				Max: math.MaxUint64,
+			},
+			MapEditors: make(map[string]*ebpf.Map),
+			ConstantEditors: []manager.ConstantEditor{
+				ddebpf.GetPatchedPrintkEditor(),
+			},
+		}
+
+		tp, err := tracefs.OpenFile("trace_pipe", os.O_RDONLY, 0)
```

Comment:
You can use our parser in `pkg/ebpf/ebpftest/trace_pipe_linux.go` instead.

Reply:
I saw it, but I preferred to have some more explicit management so that I can see clearly what's being done with newlines